### PR TITLE
Correct the init function to use initLocked() for initializing std.event.Lock.

### PR DESCRIPTION
**Pull request by kaihlavirta**
#### init funcion used init() to initialize a std.event.Lock. std.event.Lock does not have init() function. Corrected to initLocked().

### CODE REVIEW

File: lib/std/event/group.zig [hunk 1]
```diff
@@ -35,7 +35,7 @@ pub fn Group(comptime ReturnType: type) type {
             return Self{
                 .frame_stack = Stack.init(),
                 .alloc_stack = AllocStack.init(),
-                .lock = Lock.init(),
+                .lock = Lock.initLocked(),
```
Comment (Vexu) [inline lib/std/event/group.zig:38]: Looking at the commit which caused this a31d9f92f282a878836a3ecac5a48d5f4037868c, the lock is supposed to be initiated in an unlocked state:
```suggestion
                .lock = .{},
```

Also, could you apply the same change to `std.event.Locked` as mentioned in the issue?

**Review by Vexu — CHANGES_REQUESTED** [review]

**Comment by kaihlavirta** [comment]
Updated :
* lib/std/event/group.zig
* lib/std/event/locked.zig

to use `.lock = .{}` instead of `.lock = Lock.init()` or `.lock = Lock.initLocked()`

**Comment by haolian9** [comment]
should tests in [here](https://github.com/kaihlavirta/zig/blob/6b260f7968c21edad80f67b38441bb2d174ec29f/lib/std/event/group.zig#L125) be enabled?

**Comment by Vexu** [comment]
They could be enabled if they were fixed but evented IO is not being tested in the CI currently due to stage1 memory usage.

**Pull request merged**